x/evm/keeper/state_transition.go

```diff
@@ -44,6 +44,10 @@ func (k *Keeper) NewEVM(
 	}
 
 	txCtx := core.NewEVMTxContext(msg)
+
+	if tracer == nil {
+		tracer = types.NewDummyTracer()
```

Comment:
I don't understand the context for this change and how this relates to the `ApplyNativeMessage`

Reply:
It's just a little convenient for the caller that they can pass nil for a default dummy tracer.

Reply:
but isn’t `nil` also a valid option?

Reply:
No, without the dummy tracer, `nil` will cause a nil pointer dereference panic in go-ethereum somewhere, when the `debug` in chain confg is true.

Reply:
ouch that's bad. Ok let's add it then and add a comment on the `DummyTracer` then